ddl/db_test.go

```diff
@@ -1226,8 +1226,10 @@ func (s *testDBSuite) TestChangeColumn(c *C) {
 	s.testErrorCode(c, sql, tmysql.ErrWrongDBName)
 	sql = "alter table t3 change t.a aa bigint"
 	s.testErrorCode(c, sql, tmysql.ErrWrongTableName)
-	sql = "alter table t3 change aa a bigint not null"
-	s.testErrorCode(c, sql, tmysql.ErrUnknown)
+	s.mustExec(c, "create table t4 (a int default '0', b varchar(10), d int not null default '0')")
+	s.tk.MustExec("insert into t4(a) values (null);")
+	sql = "alter table t4 change a a1 int not null"
+	s.testErrorCode(c, sql, tmysql.WarnDataTruncated)
```

Comment:
Is this the same with mysql? In my environment:
```
mysql> alter table t4 change a a1 int not null;
ERROR 1138 (22004): Invalid use of NULL value
```

Reply:
@lamxTyler Done

mysql
```sql
mysql> create table test2 (c1 int, c2 int, c3 int default 1, index (c1));
Query OK, 0 rows affected (0.03 sec)

mysql> insert into test2(c2) values (null);
Query OK, 1 row affected (0.00 sec)
mysql> alter table test2 change c2 a bigint not null;
ERROR 1265 (01000): Data truncated for column 'a' at row 1
```

Reply:
Your test also changes the column type, but if you use `alter table test2 change c2 a int not null;`, you can get a different result.

Reply:
```
mysql> create table test2 (c1 int, c2 int, c3 int default 1, index (c1));
Query OK, 0 rows affected (0.01 sec)

mysql> insert into test2(c2) values (null);
Query OK, 1 row affected (0.00 sec)

mysql> alter table test2 change c2 a bigint not null;
ERROR 1265 (01000): Data truncated for column 'a' at row 1
mysql> alter table test2 change c2 a int not null;
ERROR 1138 (22004): Invalid use of NULL value
mysql> alter table test2 change c2 a int not null;
ERROR 1138 (22004): Invalid use of NULL value
```
FYI @ciscoxll